### PR TITLE
[NodeBalancer Refactor] - Redux Containers

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

Adds containers for NodeBalancers, configs, and nodes

Also adds nodes to Redux state.

## Type of Change
- Non breaking change ('update', 'change')

### CODE REVIEW

File: packages/manager/src/containers/__new__/nodeBalancers.container.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  createNodeBalancer: (
+    payload: CreateNodeBalancerPayload
+  ) => Promise<NodeBalancer>;
+  deleteNodeBalancer: (id: number) => Promise<{}>;
+  updateNodeBalancer: (
+    params: UpdateNodeBalancerParams
+  ) => Promise<NodeBalancer>;
+}
+
+/* tslint:disable-next-line */
```
Comment (Jskobos) [inline packages/manager/src/containers/__new__/nodeBalancers.container.ts]: don't think this is needed anymore

File: packages/manager/src/containers/__new__/nodeBalancerConfigs.container.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  deleteNodeBalancerConfig: (
+    configId: number,
+    nodeBalancerId: number
+  ) => Promise<{}>;
+  updateNodeBalancerConfig: (
+    params: UpdateNodeBalancerConfigParams
+  ) => Promise<NodeBalancerConfig>;
+}
+
+/* tslint:disable-next-line */
```
Comment (Jskobos) [inline packages/manager/src/containers/__new__/nodeBalancerConfigs.container.ts]: this doesn't seem to be doing anything

**Comment by martinmckenna** [comment]
@Jskobos The `__new__` business is just for my own sanity. I'll remove it all at the end.

re: the redux stuff - yeah there's _lots_ of problems with the Redux state that will have to be solved. I want to clean up the NodeBalancerConfig state and make it adopt an easier-to-use shape, so we can more easily reference a config with it's parent nodebalancer.

Fun changes coming soon.

File: packages/manager/src/containers/__new__/nodeBalancerConfigNodes.container.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    ReduxStateProps & Partial<StateProps> & DispatchProps & OwnProps,
+    OwnProps
+  >;
+  <ReduxStateProps, OwnProps>(): InferableComponentEnhancerWithProps<
+    ReduxStateProps & DispatchProps & OwnProps,
+    OwnProps
+  >;
+}
+
+const connected: Connected = <ReduxState extends {}, OwnProps extends {}>(
```
Comment (johnwcallahan) [inline packages/manager/src/containers/__new__/nodeBalancerConfigNodes.container.ts]: Minor: This is called `ReduxStateProps` in the `Connected` interface above. I'd use that name to be consistent.

File: packages/manager/src/containers/__new__/nodeBalancerConfigs.container.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    ReduxStateProps & Partial<StateProps> & DispatchProps & OwnProps,
+    OwnProps
+  >;
+  <ReduxStateProps, OwnProps>(): InferableComponentEnhancerWithProps<
+    ReduxStateProps & DispatchProps & OwnProps,
+    OwnProps
+  >;
+}
+
+const connected: Connected = <ReduxState extends {}, OwnProps extends {}>(
```
Comment (johnwcallahan) [inline packages/manager/src/containers/__new__/nodeBalancerConfigs.container.ts]: Minor: (same comment here about `ReduxStateProps`.)

File: packages/manager/src/containers/__new__/nodeBalancerConfigNodes.container.ts (hunk 3 above)
Comment (johnwcallahan) [reply packages/manager/src/containers/__new__/nodeBalancerConfigNodes.container.ts]: I'm not sure how this plays into the new data shape, so feel free to disregard if if won't matter when the data shape changes.

**Pull request merged**